Content.Server/Drone/DroneSystem.cs

```diff
@@ -0,0 +1,145 @@
+        {
+            if (args.IsInDetailsRange)
+            {
+                if (EntityManager.TryGetComponent<MindComponent>(uid, out var mind) && mind.HasMind)
+                {
+                    args.PushMarkup(Loc.GetString("drone-active"));
+                }
+                else
+                {
+                    args.PushMarkup(Loc.GetString("drone-dormant"));
+                }
+            }
+        }
+
+        private void OnMobStateChanged(EntityUid uid, DroneComponent drone, MobStateChangedEvent args)
+        {
+            if (args.Component.IsDead())
+            {
+                foreach (var item in drone.ToolUids)
+                {
+                        EntityManager.DeleteEntity(item);
+                }
+                _explosions.SpawnExplosion(uid);
+                EntityManager.DeleteEntity(uid);
+            }
+        }
+
+        private void OnDisarmAttempt(EntityUid uid, DroneComponent drone, DisarmAttemptEvent args)
+        {
+            EntityManager.TryGetComponent<HandsComponent>(args.TargetUid, out var hands);
+            var item = hands.GetActiveHandItem;
+            if (EntityManager.TryGetComponent<DroneToolComponent>(item?.Owner, out var itemInHand))
+            {
+            args.Cancel();
+            }
+        }
+
+        private void OnMindAdded(EntityUid uid, DroneComponent drone, MindAddedMessage args)
+        {
+            EntityManager.TryGetComponent<TagComponent>(uid, out var tagComp);
+            UpdateDroneAppearance(uid, DroneStatus.On);
+            tagComp.AddTag("DoorBumpOpener");
+            _popupSystem.PopupEntity(Loc.GetString("drone-activated"), uid, Filter.Pvs(uid));
+
+            if (drone.alreadyAwoken == false)
+            {
+                var spawnCoord = Transform(uid).Coordinates;
+
+                if (drone.Tools.Count == 0) return;
+
+                if (EntityManager.TryGetComponent<HandsComponent>(uid, out var hands))
+                {
+                   foreach (var entry in drone.Tools)
+                    {
+                        var item = EntityManager.SpawnEntity(entry.PrototypeId, spawnCoord);
+                        EntityManager.AddComponent<DroneToolComponent>(item);
+                        hands.PutInHand(item);
+                        drone.ToolUids.Add(item);
+                    }
+                }
+
+                drone.alreadyAwoken = true;
+            }
+        }
+
+        private void OnMindRemoved(EntityUid uid, DroneComponent drone, MindRemovedMessage args)
+        {
+            EntityManager.TryGetComponent<TagComponent>(uid, out var tagComp);
+            UpdateDroneAppearance(uid, DroneStatus.Off);
+            tagComp.RemoveTag("DoorBumpOpener");
+            EntityManager.EnsureComponent<GhostTakeoverAvailableComponent>(uid);
+        }
+
+        private void OnDropAttempt(EntityUid uid, DroneComponent drone, DropAttemptEvent args)
+        {
```

Comment:
isnt there an event raised on the item being dropped you can sub to instead on DroneToolComponent. then you dont have to check every time

Reply:
same with unequip attempt? you might only need one of these, tbh, since stopping it from being dropped should also stop it from being unequipped

Reply:
Unsure how that will behave with e.g. trying to put them in a container or something

Reply:
Also if there is an event raised on the item then I can't find it